File: Source/Entities/SampleEntity.vb
Imports Celeste.[Mod].Entities
Imports Microsoft.Xna.Framework
Imports Monocle

<CustomEntity("VeryBasicMod/SampleEntity")>
Public Class SampleEntity
    Inherits Entity
    Public Sub New(data As EntityData, offset As Vector2)
        MyBase.New(data.Position + offset)
        ' TODO: read properties from data
        Add(GFX.SpriteBank.Create("sampleEntity"))
        Collider = New Hitbox(16, 16, -8, -8)
    End Sub

End Class

File: Source/Entities/SampleEntity.fs
namespace Celeste.Mod.VeryFunnyMod.Entities

open Celeste.Mod
open Celeste
open System
open Monocle
open Microsoft.Xna.Framework
open Celeste.Mod.Entities

[<CustomEntity("VeryFunnyMod/SampleEntity")>]
type SampleEntity(data: EntityData, offset: Vector2) as this =
    inherit Entity(data.Position + offset)

    do
        this.Add(GFX.SpriteBank.Create("sampleEntity"))
        this.Collider <- new Hitbox(16f, 16f, -8f, -8f)
